### PR TITLE
MINOR: Make PushHttpMetricsReporter API compatible with releases back to 0.8.2.2

### PR DESCRIPTION
**Pull request by ewencp**
This is follow up to #4072 which added the PushHttpMetricsReporter and converted some services to use it. We somehow missed some compatibility issues that made the ProducerPerformance tool fail when using a newer tools jar with older common/clients jar, which we do with some system tests so we have all the features we need in the tool but can build compatibility tests for older releases.

This just adjusts some API usage to make the tool compatible with all previous releases.

I have a full run of the tests starting [here](https://jenkins.confluent.io/job/system-test-kafka-branch-builder/1122/)

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by asfgit** [comment]
FAILURE
 8067 tests run, 5 skipped, 2 failed.
--none--

**Comment by asfgit** [comment]
FAILURE
 8067 tests run, 5 skipped, 2 failed.
--none--

**Comment by asfgit** [comment]
SUCCESS 
 8067 tests run, 5 skipped, 0 failed.
--none--

**Comment by ijuma** [comment]
Streams failures seem unrelated.

**Review by ijuma — APPROVED** [review]
LGTM if the tests pass.

I don't seem to be able to check the checklist items, which is annoying. It worked when I tested it on my own PRs, but maybe it won't work with PRs from other people (maybe because we don't have write access to the repo).

File: tools/src/main/java/org/apache/kafka/tools/PushHttpMetricsReporter.java
Comment (ijuma) [inline tools/src/main/java/org/apache/kafka/tools/PushHttpMetricsReporter.java:320]: When are we going to drop support for 0.8.x? :)

Comment (ewencp) [reply tools/src/main/java/org/apache/kafka/tools/PushHttpMetricsReporter.java:320]: Heh, I was actually thinking about that too since we could just remove the affected tests. It was only affecting `kafkatest.sanity_checks.test_performance_services.PerformanceServiceTest` which doesn't seem that critical anyway. But since I was able to work out how to make it compatible I figured I'd leave it for now. More generally the major thing I think we evaluate for such old versions is compatibility. It would probably be a good idea to collectively agree on a reasonable range of versions (or based on time) so we can cap how far back we need to continue maintaining tests. 2 years might be a good range, in which case we'd be able to drop tests for the 0.9 line soon too.

**Comment by ewencp** [comment]
Merging to trunk and 1.0 and will update https://github.com/apache/kafka/pull/4207 with these changes as well.

**Pull request closed without merging**
